packages/core/src/sdk/search/Sentinel.tsx

```diff
@@ -38,19 +39,24 @@ const replacePagination = (page: string, router: NextRouter) => {
  * Also, this component's name is kind of curious. Wikipedia calls is Page Break(https://en.wikipedia.org/wiki/Page_break)
  * however all codes I've seen online use Sentinel
  */
-function Sentinel({ page, pageSize, products, title }: Props) {
-  const viewedRef = useRef(false)
-  const { ref, inView } = useInView()
-  const { pages } = useSearch()
-
+function Sentinel({
+  page,
+  pageSize,
+  products,
+  title,
+  children,
+}: PropsWithChildren<SentinelProps>) {
   const router = useRouter()
-
+  const { pages } = useSearch()
+  const { isDesktop } = useScreenResize()
+  const { ref, inView } = useInView({ threshold: isDesktop ? 0.7 : 0.3 })
```

Comment:
The values here were based on my testing.
The mobile version seems fine with 30%. Desktop is working well with 70%, but other screen sizes can behave differently.